### ISSUE TITLE
Crunchy PostgreSQL 5.3.0 cannot provision PostgreSQL 11 cluster

### ISSUE THREAD

**Issue by bukovjanmic** [CrunchyData/postgres-operator#3546]
Using Crunchy PostgreSQL 5.3.0 operator, it is unable to provision Postgres 11 cluster, which should be supported. Postgres 13 and 15 that we tested work fine.

The problem is that StatefulSets that gets created are missing image parameters for init and regular containers.

The is PostgreSQL CR:

```
apiVersion: postgres-operator.crunchydata.com/v1beta1
kind: PostgresCluster
metadata:
  name: postgres
  namespace: ictsritip-poc-powerflow
spec:
  backups:
    pgbackrest:
      global:
        repo1-retention-full: '2'
        repo1-retention-full-type: count
      jobs:
        resources:
          limits:
            cpu: '1'
            memory: 500Mi
          requests:
            cpu: 500m
            memory: 2Gi
      repos:
        - name: repo1
          schedules:
            differential: 0 1 * * 1-6
            full: 0 1 * * 0
          volume:
            volumeClaimSpec:
              accessModes:
                - ReadWriteOnce
              resources:
                requests:
                  storage: 20Gi
  instances:
    - affinity:
        podAntiAffinity:
          preferredDuringSchedulingIgnoredDuringExecution:
            - podAffinityTerm:
                labelSelector:
                  matchExpressions:
                    - key: postgres-operator.crunchydata.com/data
                      operator: In
                      values:
                        - postgres
                topologyKey: kubernetes.io/hostname
              weight: 100
      dataVolumeClaimSpec:
        accessModes:
          - ReadWriteOnce
        resources:
          requests:
            storage: 10Gi
      name: ''
      replicas: 2
      resources:
        limits:
          cpu: '1'
          memory: 4Gi
        requests:
          cpu: 500m
          memory: 1Gi
  monitoring:
    pgmonitor:
      exporter:
        image: >-
          central-docker-release.artif.cc.corp/crunchydata/crunchy-postgres-exporter:ubi8-5.1.1-0
  patroni:
    dynamicConfiguration:
      postgresql:
        pg_hba:
          - hostnossl all all all md5
    leaderLeaseDurationSeconds: 30
    port: 8008
    syncPeriodSeconds: 10
  port: 5432
  postgresVersion: 11
  users:
    - databases:
        - powerflow
      name: powerflow
```

This is the resulting StatefulSet for Postgres. It is missing the image parameter for (init) containers.

```
kind: StatefulSet
apiVersion: apps/v1
metadata:
  resourceVersion: '1626380339'
  name: postgres-00-69cc
  uid: 56c519fa-694c-4a07-baae-210b48a59178
  creationTimestamp: '2023-01-25T12:01:58Z'
  generation: 1
  namespace: ictsritip-poc-powerflow
  ownerReferences:
    - apiVersion: postgres-operator.crunchydata.com/v1beta1
      kind: PostgresCluster
      name: postgres
      uid: e7278847-48a6-4e1c-ace8-38e135d7df7a
      controller: true
      blockOwnerDeletion: true
  labels:
    postgres-operator.crunchydata.com/cluster: postgres
    postgres-operator.crunchydata.com/data: postgres
    postgres-operator.crunchydata.com/instance: postgres-00-69cc
    postgres-operator.crunchydata.com/instance-set: '00'
spec:
  replicas: 1
  selector:
    matchLabels:
      postgres-operator.crunchydata.com/cluster: postgres
      postgres-operator.crunchydata.com/instance: postgres-00-69cc
      postgres-operator.crunchydata.com/instance-set: '00'
  template:
    metadata:
      creationTimestamp: null
      labels:
        postgres-operator.crunchydata.com/cluster: postgres
        postgres-operator.crunchydata.com/crunchy-postgres-exporter: 'true'
        postgres-operator.crunchydata.com/data: postgres
        postgres-operator.crunchydata.com/instance: postgres-00-69cc
        postgres-operator.crunchydata.com/instance-set: '00'
        postgres-operator.crunchydata.com/patroni: postgres-ha
    spec:
      restartPolicy: Always
      initContainers:
        - resources:
            limits:
              cpu: '1'
              memory: 4Gi
            requests:
              cpu: 500m
              memory: 1Gi
          terminationMessagePath: /dev/termination-log
          name: postgres-startup
          command:
            - bash
            - '-ceu'
            - '--'
            - >-
              declare -r expected_major_version="$1" pgwal_directory="$2"
              pgbrLog_directory="$3"

              permissions() { while [[ -n "$1" ]]; do set "${1%/*}" "$@"; done;
              shift; stat -Lc '%A %4u %4g %n' "$@"; }

              halt() { local rc=$?; >&2 echo "$@"; exit "${rc/#0/1}"; }

              results() { printf '::postgres-operator: %s::%s\n' "$@"; }

              recreate() (
                local tmp; tmp=$(mktemp -d -p "${1%/*}"); GLOBIGNORE='.:..'; set -x
                chmod "$2" "${tmp}"; mv "$1"/* "${tmp}"; rmdir "$1"; mv "${tmp}" "$1"
              )

              safelink() (
                local desired="$1" name="$2" current
                current=$(realpath "${name}")
                if [ "${current}" = "${desired}" ]; then return; fi
                set -x; mv --no-target-directory "${current}" "${desired}"
                ln --no-dereference --force --symbolic "${desired}" "${name}"
              )

              echo Initializing ...

              results 'uid' "$(id -u)" 'gid' "$(id -G)"

              results 'postgres path' "$(command -v postgres)"

              results 'postgres version' "${postgres_version:=$(postgres
              --version)}"

              [[ "${postgres_version}" =~ ")
              ${expected_major_version}"($|[^0-9]) ]] ||

              halt Expected PostgreSQL version "${expected_major_version}"

              results 'config directory' "${PGDATA:?}"

              postgres_data_directory=$([ -d "${PGDATA}" ] && postgres -C
              data_directory || echo "${PGDATA}")

              results 'data directory' "${postgres_data_directory}"

              [[ "${postgres_data_directory}" == "${PGDATA}" ]] ||

              halt Expected matching config and data directories

              bootstrap_dir="${postgres_data_directory}_bootstrap"

              [ -d "${bootstrap_dir}" ] && results 'bootstrap directory'
              "${bootstrap_dir}"

              [ -d "${bootstrap_dir}" ] &&
              postgres_data_directory="${bootstrap_dir}"

              if [[ ! -e "${postgres_data_directory}" || -O
              "${postgres_data_directory}" ]]; then

              install --directory --mode=0700 "${postgres_data_directory}"

              elif [[ -w "${postgres_data_directory}" && -g
              "${postgres_data_directory}" ]]; then

              recreate "${postgres_data_directory}" '0700'

              else (halt Permissions!); fi ||

              halt "$(permissions "${postgres_data_directory}" ||:)"

              results 'pgBackRest log directory' "${pgbrLog_directory}"

              install --directory --mode=0775 "${pgbrLog_directory}" ||

              halt "$(permissions "${pgbrLog_directory}" ||:)"

              install -D --mode=0600 -t "/tmp/replication"
              "/pgconf/tls/replication"/{tls.crt,tls.key,ca.crt}

              [ -f "${postgres_data_directory}/PG_VERSION" ] || exit 0

              results 'data version' "${postgres_data_version:=$(<
              "${postgres_data_directory}/PG_VERSION")}"

              [[ "${postgres_data_version}" == "${expected_major_version}" ]] ||

              halt Expected PostgreSQL data version "${expected_major_version}"

              [[ ! -f "${postgres_data_directory}/postgresql.conf" ]] &&

              touch "${postgres_data_directory}/postgresql.conf"

              safelink "${pgwal_directory}" "${postgres_data_directory}/pg_wal"

              results 'wal directory' "$(realpath
              "${postgres_data_directory}/pg_wal")"

              rm -f "${postgres_data_directory}/recovery.signal"
            - startup
            - '11'
            - /pgdata/pg11_wal
            - /pgdata/pgbackrest/log
          env:
            - name: PGDATA
              value: /pgdata/pg11
            - name: PGHOST
              value: /tmp/postgres
            - name: PGPORT
              value: '5432'
            - name: KRB5_CONFIG
              value: /etc/postgres/krb5.conf
            - name: KRB5RCACHEDIR
              value: /tmp
          securityContext:
            capabilities:
              drop:
                - ALL
            privileged: false
            runAsNonRoot: true
            readOnlyRootFilesystem: true
            allowPrivilegeEscalation: false
          imagePullPolicy: IfNotPresent
          volumeMounts:
            - name: cert-volume
              readOnly: true
              mountPath: /pgconf/tls
            - name: postgres-data
              mountPath: /pgdata
            - name: tmp
              mountPath: /tmp
          terminationMessagePolicy: File
        - name: nss-wrapper-init
          command:
            - bash
            - '-c'
            - >
              export NSS_WRAPPER_SUBDIR=postgres CRUNCHY_NSS_USERNAME=postgres
              CRUNCHY_NSS_USER_DESC="postgres" 

              # Define nss_wrapper directory and passwd & group files that will
              be utilized by nss_wrapper.  The

              # nss_wrapper_env.sh script (which also sets these vars) isn't
              sourced here since the nss_wrapper

              # has not yet been setup, and we therefore don't yet want the
              nss_wrapper vars in the environment.

              mkdir -p /tmp/nss_wrapper

              chmod g+rwx /tmp/nss_wrapper


              NSS_WRAPPER_DIR="/tmp/nss_wrapper/${NSS_WRAPPER_SUBDIR}"

              NSS_WRAPPER_PASSWD="${NSS_WRAPPER_DIR}/passwd"

              NSS_WRAPPER_GROUP="${NSS_WRAPPER_DIR}/group"


              # create the nss_wrapper directory

              mkdir -p "${NSS_WRAPPER_DIR}"


              # grab the current user ID and group ID

              USER_ID=$(id -u)

              export USER_ID

              GROUP_ID=$(id -g)

              export GROUP_ID


              # get copies of the passwd and group files

              [[ -f "${NSS_WRAPPER_PASSWD}" ]] || cp "/etc/passwd"
              "${NSS_WRAPPER_PASSWD}"

              [[ -f "${NSS_WRAPPER_GROUP}" ]] || cp "/etc/group"
              "${NSS_WRAPPER_GROUP}"


              # if the username is missing from the passwd file, then add it

              if [[ ! $(cat "${NSS_WRAPPER_PASSWD}") =~
              ${CRUNCHY_NSS_USERNAME}:x:${USER_ID} ]]; then
                  echo "nss_wrapper: adding user"
                  passwd_tmp="${NSS_WRAPPER_DIR}/passwd_tmp"
                  cp "${NSS_WRAPPER_PASSWD}" "${passwd_tmp}"
                  sed -i "/${CRUNCHY_NSS_USERNAME}:x:/d" "${passwd_tmp}"
                  # needed for OCP 4.x because crio updates /etc/passwd with an entry for USER_ID
                  sed -i "/${USER_ID}:x:/d" "${passwd_tmp}"
                  printf '${CRUNCHY_NSS_USERNAME}:x:${USER_ID}:${GROUP_ID}:${CRUNCHY_NSS_USER_DESC}:${HOME}:/bin/bash\n' >> "${passwd_tmp}"
                  envsubst < "${passwd_tmp}" > "${NSS_WRAPPER_PASSWD}"
                  rm "${passwd_tmp}"
              else
                  echo "nss_wrapper: user exists"
              fi


              # if the username (which will be the same as the group name) is
              missing from group file, then add it

              if [[ ! $(cat "${NSS_WRAPPER_GROUP}") =~
              ${CRUNCHY_NSS_USERNAME}:x:${USER_ID} ]]; then
                  echo "nss_wrapper: adding group"
                  group_tmp="${NSS_WRAPPER_DIR}/group_tmp"
                  cp "${NSS_WRAPPER_GROUP}" "${group_tmp}"
                  sed -i "/${CRUNCHY_NSS_USERNAME}:x:/d" "${group_tmp}"
                  printf '${CRUNCHY_NSS_USERNAME}:x:${USER_ID}:${CRUNCHY_NSS_USERNAME}\n' >> "${group_tmp}"
                  envsubst < "${group_tmp}" > "${NSS_WRAPPER_GROUP}"
                  rm "${group_tmp}"
              else
                  echo "nss_wrapper: group exists"
              fi


              # export the nss_wrapper env vars

              # define nss_wrapper directory and passwd & group files that will
              be utilized by nss_wrapper

              NSS_WRAPPER_DIR="/tmp/nss_wrapper/${NSS_WRAPPER_SUBDIR}"

              NSS_WRAPPER_PASSWD="${NSS_WRAPPER_DIR}/passwd"

              NSS_WRAPPER_GROUP="${NSS_WRAPPER_DIR}/group"


              export LD_PRELOAD=/usr/lib64/libnss_wrapper.so

              export NSS_WRAPPER_PASSWD="${NSS_WRAPPER_PASSWD}"

              export NSS_WRAPPER_GROUP="${NSS_WRAPPER_GROUP}"


              echo "nss_wrapper: environment configured"
          resources:
            limits:
              cpu: '1'
              memory: 4Gi
            requests:
              cpu: 500m
              memory: 1Gi
          volumeMounts:
            - name: tmp
              mountPath: /tmp
          terminationMessagePath: /dev/termination-log
          terminationMessagePolicy: File
          imagePullPolicy: IfNotPresent
          securityContext:
            capabilities:
              drop:
                - ALL
            privileged: false
            runAsNonRoot: true
            readOnlyRootFilesystem: true
            allowPrivilegeEscalation: false
      serviceAccountName: postgres-instance
      schedulerName: default-scheduler
      enableServiceLinks: false
      affinity:
        podAntiAffinity:
          preferredDuringSchedulingIgnoredDuringExecution:
            - weight: 100
              podAffinityTerm:
                labelSelector:
                  matchExpressions:
                    - key: postgres-operator.crunchydata.com/data
                      operator: In
                      values:
                        - postgres
                topologyKey: kubernetes.io/hostname
      terminationGracePeriodSeconds: 30
      shareProcessNamespace: true
      securityContext:
        fsGroupChangePolicy: OnRootMismatch
      containers:
        - resources:
            limits:
              cpu: '1'
              memory: 4Gi
            requests:
              cpu: 500m
              memory: 1Gi
          readinessProbe:
            httpGet:
              path: /readiness
              port: 8008
              scheme: HTTPS
            initialDelaySeconds: 3
            timeoutSeconds: 5
            periodSeconds: 10
            successThreshold: 1
            failureThreshold: 3
          terminationMessagePath: /dev/termination-log
          name: database
          command:
            - patroni
            - /etc/patroni
          livenessProbe:
            httpGet:
              path: /liveness
              port: 8008
              scheme: HTTPS
            initialDelaySeconds: 3
            timeoutSeconds: 5
            periodSeconds: 10
            successThreshold: 1
            failureThreshold: 3
          env:
            - name: PGDATA
              value: /pgdata/pg11
            - name: PGHOST
              value: /tmp/postgres
            - name: PGPORT
              value: '5432'
            - name: KRB5_CONFIG
              value: /etc/postgres/krb5.conf
            - name: KRB5RCACHEDIR
              value: /tmp
            - name: PATRONI_NAME
              valueFrom:
                fieldRef:
                  apiVersion: v1
                  fieldPath: metadata.name
            - name: PATRONI_KUBERNETES_POD_IP
              valueFrom:
                fieldRef:
                  apiVersion: v1
                  fieldPath: status.podIP
            - name: PATRONI_KUBERNETES_PORTS
              value: |
                - name: postgres
                  port: 5432
                  protocol: TCP
            - name: PATRONI_POSTGRESQL_CONNECT_ADDRESS
              value: '$(PATRONI_NAME).postgres-pods:5432'
            - name: PATRONI_POSTGRESQL_LISTEN
              value: '*:5432'
            - name: PATRONI_POSTGRESQL_CONFIG_DIR
              value: /pgdata/pg11
            - name: PATRONI_POSTGRESQL_DATA_DIR
              value: /pgdata/pg11
            - name: PATRONI_RESTAPI_CONNECT_ADDRESS
              value: '$(PATRONI_NAME).postgres-pods:8008'
            - name: PATRONI_RESTAPI_LISTEN
              value: '*:8008'
            - name: PATRONICTL_CONFIG_FILE
              value: /etc/patroni
            - name: LD_PRELOAD
              value: /usr/lib64/libnss_wrapper.so
            - name: NSS_WRAPPER_PASSWD
              value: /tmp/nss_wrapper/postgres/passwd
            - name: NSS_WRAPPER_GROUP
              value: /tmp/nss_wrapper/postgres/group
          securityContext:
            capabilities:
              drop:
                - ALL
            privileged: false
            runAsNonRoot: true
            readOnlyRootFilesystem: true
            allowPrivilegeEscalation: false
          ports:
            - name: postgres
              containerPort: 5432
              protocol: TCP
          imagePullPolicy: IfNotPresent
          volumeMounts:
            - name: cert-volume
              readOnly: true
              mountPath: /pgconf/tls
            - name: postgres-data
              mountPath: /pgdata
            - name: database-containerinfo
              readOnly: true
              mountPath: /etc/database-containerinfo
            - name: pgbackrest-config
              readOnly: true
              mountPath: /etc/pgbackrest/conf.d
            - name: patroni-config
              readOnly: true
              mountPath: /etc/patroni
            - name: tmp
              mountPath: /tmp
            - name: dshm
              mountPath: /dev/shm
          terminationMessagePolicy: File
        - name: replication-cert-copy
          command:
            - bash
            - '-ceu'
            - '--'
            - |-
              monitor() {
              declare -r directory="/pgconf/tls"
              exec {fd}<> <(:)
              while read -r -t 5 -u "${fd}" || true; do
                if [ "${directory}" -nt "/proc/self/fd/${fd}" ] &&
                  install -D --mode=0600 -t "/tmp/replication" "${directory}"/{replication/tls.crt,replication/tls.key,replication/ca.crt} &&
                  pkill -HUP --exact --parent=1 postgres
                then
                  exec {fd}>&- && exec {fd}<> <(:)
                  stat --format='Loaded certificates dated %y' "${directory}"
                fi
              done
              }; export -f monitor; exec -a "$0" bash -ceu monitor
            - replication-cert-copy
          resources: {}
          volumeMounts:
            - name: cert-volume
              readOnly: true
              mountPath: /pgconf/tls
            - name: tmp
              mountPath: /tmp
          terminationMessagePath: /dev/termination-log
          terminationMessagePolicy: File
          imagePullPolicy: IfNotPresent
          securityContext:
            capabilities:
              drop:
                - ALL
            privileged: false
            runAsNonRoot: true
            readOnlyRootFilesystem: true
            allowPrivilegeEscalation: false
        - resources: {}
          terminationMessagePath: /dev/termination-log
          name: pgbackrest
          command:
            - pgbackrest
            - server
          livenessProbe:
            exec:
              command:
                - pgbackrest
                - server-ping
            timeoutSeconds: 1
            periodSeconds: 10
            successThreshold: 1
            failureThreshold: 3
          env:
            - name: LD_PRELOAD
              value: /usr/lib64/libnss_wrapper.so
            - name: NSS_WRAPPER_PASSWD
              value: /tmp/nss_wrapper/postgres/passwd
            - name: NSS_WRAPPER_GROUP
              value: /tmp/nss_wrapper/postgres/group
          securityContext:
            capabilities:
              drop:
                - ALL
            privileged: false
            runAsNonRoot: true
            readOnlyRootFilesystem: true
            allowPrivilegeEscalation: false
          imagePullPolicy: IfNotPresent
          volumeMounts:
            - name: pgbackrest-server
              readOnly: true
              mountPath: /etc/pgbackrest/server
            - name: postgres-data
              mountPath: /pgdata
            - name: pgbackrest-config
              readOnly: true
              mountPath: /etc/pgbackrest/conf.d
            - name: tmp
              mountPath: /tmp
          terminationMessagePolicy: File
          image: >-
            registry.connect.redhat.com/crunchydata/crunchy-pgbackrest@sha256:cbeb0123e7439b12bece249b730eb367fcf5c834541273497c37057deedcdc58
        - resources: {}
          terminationMessagePath: /dev/termination-log
          name: pgbackrest-config
          command:
            - bash
            - '-ceu'
            - '--'
            - >-
              monitor() {

              exec {fd}<> <(:)

              until read -r -t 5 -u "${fd}"; do
                if
                  [ "${filename}" -nt "/proc/self/fd/${fd}" ] &&
                  pkill -HUP --exact --parent=0 pgbackrest
                then
                  exec {fd}>&- && exec {fd}<> <(:)
                  stat --dereference --format='Loaded configuration dated %y' "${filename}"
                elif
                  { [ "${directory}" -nt "/proc/self/fd/${fd}" ] ||
                    [ "${authority}" -nt "/proc/self/fd/${fd}" ]
                  } &&
                  pkill -HUP --exact --parent=0 pgbackrest
                then
                  exec {fd}>&- && exec {fd}<> <(:)
                  stat --format='Loaded certificates dated %y' "${directory}"
                fi
              done

              }; export directory="$1" authority="$2" filename="$3"; export -f
              monitor; exec -a "$0" bash -ceu monitor
            - pgbackrest-config
            - /etc/pgbackrest/server
            - /etc/pgbackrest/conf.d/~postgres-operator/tls-ca.crt
            - /etc/pgbackrest/conf.d/~postgres-operator_server.conf
          securityContext:
            capabilities:
              drop:
                - ALL
            privileged: false
            runAsNonRoot: true
            readOnlyRootFilesystem: true
            allowPrivilegeEscalation: false
          imagePullPolicy: IfNotPresent
          volumeMounts:
            - name: pgbackrest-server
              readOnly: true
              mountPath: /etc/pgbackrest/server
            - name: pgbackrest-config
              readOnly: true
              mountPath: /etc/pgbackrest/conf.d
            - name: tmp
              mountPath: /tmp
          terminationMessagePolicy: File
          image: >-
            registry.connect.redhat.com/crunchydata/crunchy-pgbackrest@sha256:cbeb0123e7439b12bece249b730eb367fcf5c834541273497c37057deedcdc58
        - resources: {}
          terminationMessagePath: /dev/termination-log
          name: exporter
          command:
            - /opt/cpm/bin/start.sh
          env:
            - name: CONFIG_DIR
              value: /opt/cpm/conf
            - name: POSTGRES_EXPORTER_PORT
              value: '9187'
            - name: PGBACKREST_INFO_THROTTLE_MINUTES
              value: '10'
            - name: PG_STAT_STATEMENTS_LIMIT
              value: '20'
            - name: PG_STAT_STATEMENTS_THROTTLE_MINUTES
              value: '-1'
            - name: EXPORTER_PG_HOST
              value: localhost
            - name: EXPORTER_PG_PORT
              value: '5432'
            - name: EXPORTER_PG_DATABASE
              value: postgres
            - name: EXPORTER_PG_USER
              value: ccp_monitoring
            - name: EXPORTER_PG_PASSWORD
              valueFrom:
                secretKeyRef:
                  name: postgres-monitoring
                  key: password
          securityContext:
            capabilities:
              drop:
                - ALL
            privileged: false
            runAsNonRoot: true
            readOnlyRootFilesystem: true
            allowPrivilegeEscalation: false
          ports:
            - name: exporter
              containerPort: 9187
              protocol: TCP
          imagePullPolicy: IfNotPresent
          volumeMounts:
            - name: exporter-config
              mountPath: /conf
            - name: tmp
              mountPath: /tmp
          terminationMessagePolicy: File
          image: >-
            central-docker-release.artif.cc.corp/crunchydata/crunchy-postgres-exporter:ubi8-5.1.1-0
      topologySpreadConstraints:
        - maxSkew: 1
          topologyKey: kubernetes.io/hostname
          whenUnsatisfiable: ScheduleAnyway
          labelSelector:
            matchLabels:
              postgres-operator.crunchydata.com/cluster: postgres
            matchExpressions:
              - key: postgres-operator.crunchydata.com/data
                operator: In
                values:
                  - postgres
                  - pgbackrest
        - maxSkew: 1
          topologyKey: topology.kubernetes.io/zone
          whenUnsatisfiable: ScheduleAnyway
          labelSelector:
            matchLabels:
              postgres-operator.crunchydata.com/cluster: postgres
            matchExpressions:
              - key: postgres-operator.crunchydata.com/data
                operator: In
                values:
                  - postgres
                  - pgbackrest
      serviceAccount: postgres-instance
      volumes:
        - name: cert-volume
          projected:
            sources:
              - secret:
                  name: postgres-cluster-cert
                  items:
                    - key: tls.crt
                      path: tls.crt
                    - key: tls.key
                      path: tls.key
                    - key: ca.crt
                      path: ca.crt
              - secret:
                  name: postgres-replication-cert
                  items:
                    - key: tls.crt
                      path: replication/tls.crt
                    - key: tls.key
                      path: replication/tls.key
                    - key: ca.crt
                      path: replication/ca.crt
            defaultMode: 384
        - name: postgres-data
          persistentVolumeClaim:
            claimName: postgres-00-69cc-pgdata
        - name: database-containerinfo
          downwardAPI:
            items:
              - path: cpu_limit
                resourceFieldRef:
                  containerName: database
                  resource: limits.cpu
                  divisor: 1m
              - path: cpu_request
                resourceFieldRef:
                  containerName: database
                  resource: requests.cpu
                  divisor: 1m
              - path: mem_limit
                resourceFieldRef:
                  containerName: database
                  resource: limits.memory
                  divisor: 1Mi
              - path: mem_request
                resourceFieldRef:
                  containerName: database
                  resource: requests.memory
                  divisor: 1Mi
              - path: labels
                fieldRef:
                  apiVersion: v1
                  fieldPath: metadata.labels
              - path: annotations
                fieldRef:
                  apiVersion: v1
                  fieldPath: metadata.annotations
            defaultMode: 420
        - name: pgbackrest-server
          projected:
            sources:
              - secret:
                  name: postgres-00-69cc-certs
                  items:
                    - key: pgbackrest-server.crt
                      path: server-tls.crt
                    - key: pgbackrest-server.key
                      path: server-tls.key
                      mode: 384
            defaultMode: 420
        - name: pgbackrest-config
          projected:
            sources:
              - configMap:
                  name: postgres-pgbackrest-config
                  items:
                    - key: pgbackrest_instance.conf
                      path: pgbackrest_instance.conf
                    - key: config-hash
                      path: config-hash
                    - key: pgbackrest-server.conf
                      path: ~postgres-operator_server.conf
              - secret:
                  name: postgres-pgbackrest
                  items:
                    - key: pgbackrest.ca-roots
                      path: ~postgres-operator/tls-ca.crt
                    - key: pgbackrest-client.crt
                      path: ~postgres-operator/client-tls.crt
                    - key: pgbackrest-client.key
                      path: ~postgres-operator/client-tls.key
                      mode: 384
                  optional: true
            defaultMode: 420
        - name: patroni-config
          projected:
            sources:
              - configMap:
                  name: postgres-config
                  items:
                    - key: patroni.yaml
                      path: ~postgres-operator_cluster.yaml
              - configMap:
                  name: postgres-00-69cc-config
                  items:
                    - key: patroni.yaml
                      path: ~postgres-operator_instance.yaml
              - secret:
                  name: postgres-00-69cc-certs
                  items:
                    - key: patroni.ca-roots
                      path: ~postgres-operator/patroni.ca-roots
                    - key: patroni.crt-combined
                      path: ~postgres-operator/patroni.crt+key
            defaultMode: 420
        - name: exporter-config
          projected:
            sources: null
            defaultMode: 420
        - name: tmp
          emptyDir:
            sizeLimit: 16Mi
        - name: dshm
          emptyDir:
            medium: Memory
      dnsPolicy: ClusterFirst
  serviceName: postgres-pods
  podManagementPolicy: OrderedReady
  updateStrategy:
    type: OnDelete
  revisionHistoryLimit: 0
status:
  replicas: 0
  availableReplicas: 0
```

**Comment by benjaminjb** [comment]
Hello, hope you don't mind, I edited your comment to format those two big yaml blocks.

As for your question, we only maintain the latest two Postgres versions within our developer portal. If you're referring to upstream community Postgres, yes it is currently still receiving updates and patches from the upstream PostgreSQL community. 

We do have support for Postgres 11 available for our customers in our the Crunchy Data access portal, but not in our developer portal, which is why your `image` fields are ending up blank. 

Would be happy to share more details on what subscription builds look like if you need more access longer term to these older versions. The best place to reach out to would be [info@crunchydata.com](mailto:info@crunchydata.com) and someone on the team can coordinate to discuss in more detail.

**Comment by bukovjanmic** [comment]
Thanks for clarification.

Since the generated StatefulSet is invalid (it is not a blank image, but missing a required image parameter), maybe a better option would be - in this case - to throw an error with an explanation in the PostgresCluster CR.

I will reach out to the mentioned address to learn more about your subscriptions.

**Comment by dsessler7** [comment]
I agree that there should probably be an error in the PostgresCluster CR in this scenario, so I created a story in our dev backlog.

That being said, if you run into an issue like this in the future, I would recommend doing a `kubectl get all -n <your_namespace>` and looking closely at the various objects. If any objects have irregular statuses or don't show a full count of `READY` pods (0/1, 1/2, etc), doing a `kubectl describe <object-name>` on those objects will likely turn up some useful information. For example, in this scenario, doing a `describe` on the unready StatefulSet would have revealed an error complaining that the `image` was not set.

**Comment by pere3** [comment]
you could support pg11 yourself if you'll provide an env called RELATED_IMAGE_POSTGRES_11 to your crunchy-operator deployment

image can be build from official https://github.com/CrunchyData/crunchy-containers repo

**Comment by pere3** [comment]
wow, i just stumbled upon inability to do so myself, even with env RELATED_IMAGE_POSTGRES_11 declared - pg13 cannot be deployed with pgo 5.3.2

image is set to null, i'm pretty sure it all worked on 5.2.0 when i first tried this hack

**Comment by ValClarkson** [comment]
HI @bukovjanmic 

A fix to prevent empty RELATED_IMAGE tags from breaking postgres cluster deployments is now available in v5.4.1
